assets/js-maps.js

```diff
@@ -1,154 +1,348 @@
+    const result = dimensions.width && dimensions.height
+      ? [dimensions.width, dimensions.height]
+      : [MapConfig.iconWidth, MapConfig.iconHeight];
 
-    this.elements();
-    this.events();
-  }
+    this.cache.iconDimensions.set(iconSelector, result); // Cache the result for future use
 
-  elements() {
-    this.maps = this.block.querySelectorAll(this.selector.map);
-    this.links = this.block.querySelectorAll(this.selector.link);
-  }
+    return result;
+  },
 
-  events() {
-    this.createMap();
-  }
+  updateMapLinks(locationData) {
+    const links = document.querySelectorAll(MapConfig.selector.link);
+    if (!links || !links.length) return;
 
-  createMap() {
-    if (!this.maps.length) return false;
+    const [lat, lon] = [locationData[0].lat, locationData[0].lon],
+          mapUrl = `https://www.openstreetmap.org/?mlat=${lat}&amp;mlon=${lon}#map=${MapConfig.defaultZoom}/${lat}/${lon}`;
 
-    this.maps.forEach(map => {
-      const id = map.getAttribute(this.attr.id),
-            address = map.getAttribute(this.attr.address),
-            url = `https://nominatim.openstreetmap.org/search?q=${encodeURIComponent(address)}&format=json`;
+    this.cache.mapLinks.push(mapUrl);
 
-      const getData = async () => {
-        try {
-          const res = await fetch(url);
-          return await res.json();
-        } catch (error) {
-          console.error(error);
+    $.batchDOM(() => {
+      links.forEach((link, i) => {
+        if (this.cache.mapLinks[i]) {
+          link.setAttribute(MapConfig.attribute.href, this.cache.mapLinks[i]);
         }
-      };
-
-      const renderMap = (obj) => {
-        const icon = document.querySelector(this.selector.icon),
-              iconWidth = icon?.getBoundingClientRect().width,
-              iconHeight = icon?.getBoundingClientRect().height,
-              iconSize = iconWidth && iconHeight ? [iconWidth, iconHeight] : [40, 55],
-              lon = obj[0]?.lon,
-              lat = obj[0]?.lat;
-
-        const layers = [
-          new ol.layer.Tile({
-            source: new ol.source.OSM()
-          }),
-          new ol.layer.Vector({
-            source: new ol.source.Vector({
-              features: [
-                new ol.Feature({
-                  geometry: new ol.geom.Point(ol.proj.fromLonLat([lon, lat]))
-                })
-              ]
-            }),
-            style: new ol.style.Style({
-              image: new ol.style.Icon({
-                anchor: [0.5, 60],
-                anchorXUnits: 'fraction',
-                anchorYUnits: 'pixels',
-                imgSize: iconSize,
-                img: icon
+      })
+    })
+  },
+
+  isElementVisible(element) {
+    if (!element) return false;
+    if ($.inViewport(element)) return true;
+
+    const tabContent = element.closest(MapConfig.selector.tabsContent);
+    if (tabContent && window.getComputedStyle(tabContent).display !== 'none') return true;
+
+    return false;
```

Comment:
you could return directly `return tabContent && window.getComputedStyle(tabContent).display !== 'none')`